STEER/STEERBase/AliOADBContainer.cxx

```diff
@@ -324,90 +353,94 @@ void AliOADBContainer::WriteToFile(const char* fname) const
+  AliDebug(5,Form("File name expanded to %s",tmp.Data()));
   TFile* file(0);
   // Try to get the file from the list of already open files
   const TSeqCollection *listOfFiles(gROOT->GetListOfFiles());
-  if(listOfFiles){
-    file =dynamic_cast<TFile*> (listOfFiles->FindObject(fname));
+  if (listOfFiles) {
+    file =dynamic_cast<TFile*> (listOfFiles->FindObject(tmp.Data()));
   }
-  if(file){
+  if (file){
     AliDebug(5,"Success! File was already open!\n");
-  }
-  else{
+  } else {
     AliDebug(5,"Couldn't find file, opening it\n");
     if(TString(fname).Contains("alien://") && ! gGrid)
       TGrid::Connect("alien://");
     file = TFile::Open(fname);
   }
-  // Delete pointer from ExpandPathName()
-  delete[] fname;
-  fname=0;
+  if (!file) 
+    return 1;
   
-    // Initialize object from file
-    if (!file) return (1);
-    AliOADBContainer* cont  = 0;
-    file->GetObject(key, cont);
-    if (!cont)
-    {
-      AliError(Form("Object (%s) not found in file \n", GetName()));	
-	return 1;
-    }
+  // Initialize object from file
+  AliOADBContainer* cont  = 0;
+  file->GetObject(key, cont);
+  if (!cont) {
+    AliError(Form("Object (%s) not found in file \n", GetName()));	
+    return 1;
+  }
 
-    SetName(cont->GetName());
-    SetTitle(cont->GetTitle());
+  SetName(cont->GetName());
+  SetTitle(cont->GetTitle());
 
-    fEntries = cont->GetNumberOfEntries();
-    fLowerLimits.Set(fEntries);
-    fUpperLimits.Set(fEntries);
-    if(fEntries > fArray->GetSize()) fArray->Expand(fEntries);
-    if (!fPassNames) fPassNames = new TObjArray(100);
-    if(fEntries > fPassNames->GetSize()) fPassNames->Expand(fEntries);
+  fEntries = cont->GetNumberOfEntries();
+  fLowerLimits.Set(fEntries);
+  fUpperLimits.Set(fEntries);
+  if (fEntries > fArray->GetSize()) 
+    fArray->Expand(fEntries);
+  if (!fPassNames) 
+    fPassNames = new TObjArray(100);
+  if (fEntries > fPassNames->GetSize()) 
+    fPassNames->Expand(fEntries);
 
-    for (Int_t i = 0; i < fEntries; i++) {
-	fLowerLimits[i] = cont->LowerLimit(i); 
-	fUpperLimits[i] = cont->UpperLimit(i);
-	fArray->AddAt(cont->GetObjectByIndex(i), i);
-	TObject* passName = cont->GetPassNameByIndex(i);
-	fPassNames->AddAt(passName ? passName : new TObjString(""), i);
-    }
-    if (!fDefaultList) fDefaultList = new TList(); 
-    TIter next(cont->GetDefaultList());
-    TObject* obj;
-    while((obj = next())) fDefaultList->Add(obj);
+  for (Int_t i = 0; i < fEntries; i++) {
+    fLowerLimits[i] = cont->LowerLimit(i); 
+    fUpperLimits[i] = cont->UpperLimit(i);
+    fArray->AddAt(cont->GetObjectByIndex(i), i);
+    TObject* passName = cont->GetPassNameByIndex(i);
+    fPassNames->AddAt(passName ? passName : new TObjString(""), i);
+  }
 
-    return 0;
-    
+  if (!fDefaultList) 
+    fDefaultList = new TList(); 
+  TIter next(cont->GetDefaultList());
+  TObject* obj;
+  while((obj = next())) 
+    fDefaultList->Add(obj);
+
+  //not yet enabled
+  //SetOwner(); 
```

Comment:
Small question - is this still supposed to be commented out?

Reply:
@raymondEhlers this might break existing containers, as we saw with some of the EMCAL OADB containers. It has to be switch on by request until it is clear all containers are compatible with this.

Reply:
Okay, thanks!